File: Cpp/lcp_50.cpp
// https://leetcode.cn/problems/WHnhjV/

class Solution {
public:
    int giveGem(vector<int>& gem, vector<vector<int>>& operations) {
        for (auto &op: operations) {
            int n = gem[op[0]] / 2;
            gem[op[0]] -= n;
            gem[op[1]] += n;
        }
        int min_ = gem[0], max_ = gem[0];
        for (auto g: gem) {
            min_ = min(min_, g);
            max_ = max(max_, g);
        }
        return max_ - min_;
    }
};

File: Python3/lcp_50.py
# https://leetcode.cn/problems/WHnhjV/

class Solution:
    def giveGem(self, gem, operations):
        """
        Inputs:
            gem: list[int]
            operations: list[list[int]]
        Outputs:
            res: int
        """
        for op1, op2 in operations:
            n = gem[op1] // 2
            gem[op1] -= n
            gem[op2] += n
        min_, max_ = gem[0], gem[0]
        for g in gem:
            min_ = min(min_, g)
            max_ = max(max_, g)
        return max_ - min_